addons/website_slides/views/website_slides_templates.xml

```diff
@@ -108,12 +105,28 @@
                 <div class="float-left col-lg-8 col-md-12 col-12 mt16 mb8">
                     <h3 class="mt0">
                         <a t-attf-href="/slides/#{slug(channel)}" t-esc="channel.name"/>
+                        <span t-if="not channel.is_member" class="o_wslides_join_channel">
+                            <a role="button" class="btn btn-primary o_wslides_join_channel_link"
+                                title="Join Channel" aria-label="Join Channel" href="#"
+                                t-att-data-channel-id="channel.id">
+                                <i class="fa fa-chevron-right"></i> Join
+                            </a>
+                        </span>
+                        <span t-if="channel.is_member">
+                            <a role="button" class="btn btn-primary o_wslides_join_channel_link"
+                                title="Join Channel" aria-label="Join Channel" href="#">
+                                <i class="fa fa-chevron-right"></i> Joined
+                            </a>
+                        </span>
                         <t t-if="category">
                             / <a t-attf-href="/slides/#{slug(channel)}/category/#{slug(category)}" t-esc="category.name"/>
                         </t>
-                        <!-- Public user has not right to create/write on slide -->
-                        <t t-if="channel.can_see and channel.can_upload and not is_public_user">
-                            <a role="button" class="btn btn-primary oe_slide_js_upload" title="Upload Presentation" t-attf-channel_id="#{channel.id}" aria-label="Upload Presentation" href="#">
+                        <t t-if="channel.can_upload">
+                            <a role="button" class="btn btn-primary oe_slide_js_upload" title="Upload Presentation"
+                                aria-label="Upload Presentation" href="#"
+                                t-att-data-channel_id="channel.id"
```

Comment:
Mixing "-" and "_", not a big deal but meh.

Reply:
Actually I just splitted the line into several one. I agree with you but as said, those templates will be reworked.